shinhye/routes/postRouter.js

```diff
@@ -0,0 +1,12 @@
+const express = require("express");
+const router = express.Router();
+const postController = require("../controllers/postController");
+
+router.post("/posts", postController.createPost);
+router.patch("/posts/:postId", postController.updatePost);
+router.get("/posts", postController.viewPost);
+router.get("/posts/:userId", postController.searchPost);
+router.delete("/posts/:postId", postController.deletePost);
+router.patch("/posts/:postId", postController.likePost);
```

Comment:
RESTful API를 구현하기 위한 엔드포인트 설계가 잘못된 것 같습니다. 

이미 routes/index.js 파일에서 posts 리소르와 관련된 엔드포인트는 `127.0.0.1:3000/posts...`로 설정이 되어 있습니다. 그렇기 때문에 다시 `/posts` 리소스를 명시할 필요가 없습니다.

지금 엔드포인트는 다음과 같아 리소스의 중복이 발생하고 있는 것입니다:
`127.0.0.1:3000/posts/posts/...`

따라서 해당 router에서는 posts를 지워주세요!
```suggestion
router.post("/", postController.createPost);
router.patch("/:postId", postController.updatePost);
router.get("/", postController.viewPost);
router.get("/:userId", postController.searchPost);
router.delete("/:postId", postController.deletePost);
router.patch("/:postId", postController.likePost);
```